core/domain/stats_jobs_continuous.py

```diff
@@ -545,56 +543,49 @@ def _encode(item):
 
         if InteractionAnswerSummariesMRJobManager._entity_created_before_job_queued( # pylint: disable=line-too-long
                 item):
-            state_answers_dict = {
-                'exploration_id': item.exploration_id,
-                'exploration_version': item.exploration_version,
-                'state_name': _encode(item.state_name),
-                'interaction_id': item.interaction_id,
-                'submitted_answer_list': item.submitted_answer_list,
-            }
-
             # Output answers submitted to the exploration for this exp version.
-            versioned_key = u'%s:%s:%s' % (
-                item.exploration_id, item.exploration_version, item.state_name)
-            yield (versioned_key.encode('utf-8'),
-                   copy.deepcopy(state_answers_dict))
+            versioned_key = u'%s:%s:%s:%s' % (
+                item.exploration_id, item.exploration_version,
+                item.interaction_id, item.state_name)
+            yield (versioned_key.encode('utf-8'), item.id)
 
             # Output the same set of answers independent of the version. This
             # allows the reduce step to aggregate answers across all
             # exploration versions.
-            state_answers_dict['exploration_version'] = VERSION_ALL
-            all_versions_key = u'%s:%s:%s' % (
-                item.exploration_id, VERSION_ALL, item.state_name)
-            yield (all_versions_key.encode('utf-8'), state_answers_dict)
+            all_versions_key = u'%s:%s:%s:%s' % (
+                item.exploration_id, VERSION_ALL, item.interaction_id,
+                item.state_name)
+            yield (all_versions_key.encode('utf-8'), item.id)
 
     @staticmethod
-    def reduce(key, stringified_state_answers_dicts):
-        exploration_id, exploration_version, state_name = key.split(':')
-        interaction_id = ast.literal_eval(
-            stringified_state_answers_dicts[0])['interaction_id']
+    def reduce(key, stringified_item_ids):
+        exploration_id, exploration_version, interaction_id, state_name = (
```

Comment:
Just to check, can the state name contain colons?

Also, is exploration_version _supposed_ to be a string?